examples/audio/stft.py

```diff
@@ -0,0 +1,433 @@
+    num_samples = int(len(wav) * target_sr / sr)  # resample to 16 kHz
+    wav = resample(wav, num_samples)
+    return wav[:, None]  # Add a channel dimension (of size 1)
+
+
+"""
+Create a function that uses the `STFTSpectrogram` to compute a spectrogram,
+then plots it.
+"""
+
+
+def plot_single_spectrogram(sample_wav_data):
+    spectrogram = layers.STFTSpectrogram(
+        mode="log",
+        frame_length=SAMPLE_RATE * 20 // 1000,
+        frame_step=SAMPLE_RATE * 5 // 1000,
+        fft_length=1024,
+        trainable=False,
+    )(sample_wav_data[None, ...]).numpy()[0]
+
+    # Plot the spectrogram
+    plt.imshow(spectrogram.T, origin="lower")
+    plt.title("Single Channel Spectrogram")
+    plt.xlabel("Time")
+    plt.ylabel("Frequency")
+    plt.show()
+
+
+"""
+Create a function that uses the `STFTSpectrogram` to compute three
+spectrograms with multiple bandwidths, then aligns them as an image
+with different channels, to get a multi-bandwith spectrogram,
+then plots the spectrogram.
+"""
+
+
+def plot_multi_bandwidth_spectrogram(sample_wav_data):
+    # All spectrograms must use the same `fft_length`, `frame_step`, and
+    # `padding="same"` in order to produce spectrograms with identical shapes,
+    # hence aligning them together. `expand_dims` ensures that the shapes are
+    # compatible with image models.
+
+    spectrograms = np.concatenate(
+        [
+            layers.STFTSpectrogram(
+                mode="log",
+                frame_length=SAMPLE_RATE * x // 1000,
+                frame_step=SAMPLE_RATE * 5 // 1000,
+                fft_length=1024,
+                padding="same",
+                expand_dims=True,
+            )(sample_wav_data[None, ...]).numpy()[0, ...]
+            for x in [5, 10, 20]
+        ],
+        axis=-1,
+    ).transpose([1, 0, 2])
+
+    # normalize each color channel for better viewing
+    mn = spectrograms.min(axis=(0, 1), keepdims=True)
+    mx = spectrograms.max(axis=(0, 1), keepdims=True)
+    spectrograms = (spectrograms - mn) / (mx - mn)
+
+    plt.imshow(spectrograms, origin="lower")
+    plt.title("Multi-bandwidth Spectrogram")
+    plt.xlabel("Time")
+    plt.ylabel("Frequency")
+    plt.show()
+
+
+"""
+Demonstrate a sample wav file.
+"""
+
+sample_wav_data = read_wav_file(pd_data["filename"].tolist()[52])
+plt.plot(sample_wav_data[:, 0])
+plt.show()
+
+"""
+Plot a Spectrogram
+"""
+
+plot_single_spectrogram(sample_wav_data)
+
+"""
+Plot a multi-bandwidth spectrogram
+"""
+
+plot_multi_bandwidth_spectrogram(sample_wav_data)
+
+"""
+### Define functions to construct a TF Dataset
+"""
+
+
+def read_dataset(df, folds, batch_size, shuffle=True):
+    msk = df["fold"].isin(folds)
+    filenames = df["filename"][msk]
+    targets = df["target"][msk].values
+    waves = np.array(
+        [read_wav_file(fil) for fil in filenames], dtype=np.float32
+    )
+    ds = tf.data.Dataset.from_tensor_slices((waves, targets))
```

Comment:
Since the whole dataset is loaded in memory, I recommend not using tf.data and instead just calling `fit(waves, targets, batch_size=...)